numpy/_core/src/multiarray/array_method.c

```diff
@@ -342,13 +342,13 @@ fill_arraymethod_from_slots(
                     return -1;
                 }
             }
-            if (i >= meth->nin && NPY_DT_is_parametric(res->dtypes[i])) {
-                PyErr_Format(PyExc_TypeError,
-                        "must provide a `resolve_descriptors` function if any "
-                        "output DType is parametric. (method: %s)",
-                        spec->name);
-                return -1;
-            }
+            // if (i >= meth->nin && NPY_DT_is_parametric(res->dtypes[i])) {
+            //     PyErr_Format(PyExc_TypeError,
+            //             "must provide a `resolve_descriptors` function if any "
+            //             "output DType is parametric. (method: %s)",
+            //             spec->name);
+            //     return -1;
+            // }
```

Comment:
This should not be merged!!! I noticed that we actually cannot skip `resolve_descriptors` for any parametric dtype, at least for sorts, because the output is parametric. Should we special-case somehow?

Reply:
Yes, it's because the output is essentially fake for sorts (we enforce `data[0] == data[1]`. Do you mean we can check if that is the case here and only error if it's not? If so makes sense -- I'll do that, thank you!

Reply:
Sorry, I just realized this is initialization, so we don't actually have the descriptors. Not sure what the condition could be...